deb/publish.go

```diff
@@ -1323,14 +1324,20 @@ func (collection *PublishedRepoCollection) Remove(publishedStorageProvider aptly
 	collection.list[len(collection.list)-1], collection.list[repoPosition], collection.list =
 		nil, collection.list[len(collection.list)-1], collection.list[:len(collection.list)-1]
 
-	if !skipCleanup && len(cleanComponents) > 0 {
-		err = collection.CleanupPrefixComponentFiles(repo.Prefix, cleanComponents,
-			publishedStorageProvider.GetPublishedStorage(storage), collectionFactory, progress)
-		if err != nil {
-			if !force {
-				return fmt.Errorf("cleanup failed, use -force-drop to override: %s", err)
+	if !skipCleanup {
```

Comment:
is this just reformatting ?

Reply:
No, `len(cleanComponents) > 0` is moved into separate if/else

Reply:
ah, then it is not functional change. please only change what is needed for the PR. this lowers the coverage otherwise

Reply:
Please look closer into the patch set. It is needed for the pr...

My added lines https://github.com/aptly-dev/aptly/pull/1381/files#diff-513f82b0090aefbc689e6e0c921004c390c5e8bb8e4240ee133fedcf40cca83dR1338-R1340 should be executed if `skipCleanup` is false but even if `len(cleanComponents)` == 0.

Thus the change.

Reply:
thanks for explaining, that is why i was asking

Reply:
Feel free to mark thread as resolved if you're okay with changes.

Reply:
if you take the publishedStorage before the if block, the nested ifs are not needed, right ?

Reply:
nope.

See this block on master:

```cpp
if !skipCleanup && len(cleanComponents) > 0 {
		err = collection.CleanupPrefixComponentFiles(repo.Prefix, cleanComponents,
			publishedStorageProvider.GetPublishedStorage(storage), collectionFactory, progress)
		if err != nil {
			if !force {
				return fmt.Errorf("cleanup failed, use -force-drop to override: %s", err)
			}
		}
	}
```

Now in this PR:

```
	if !skipCleanup {
		publishedStorage := publishedStorageProvider.GetPublishedStorage(storage)
		if len(cleanComponents) > 0 {
			err = collection.CleanupPrefixComponentFiles(repo.Prefix, cleanComponents, publishedStorage, collectionFactory, progress)
			if err != nil {
				if !force {
					return fmt.Errorf("cleanup failed, use -force-drop to override: %s", err)
				}
			}
		}

		if localStorage, ok := publishedStorage.(*files.PublishedStorage); ok {
			localStorage.CleanupPublishTree(repo.Prefix, progress)
		}
	}
```

original lines moved into separate `if len(cleanComponents) > 0` block. My additions are next to it, not inside.

Both parts are enclossed in `if !skipCleanup`

Reply:
got it